Nested property interpolation in distributionManagement fails validation in Maven 4

## Summary

Maven 4 rejects POMs that use chained property references in `<distributionManagement>` IDs. The first level of interpolation resolves, but the second level is still a `${...}` expression when the validator runs, causing a failure.

## Reproduction

Build [apache/ratis](https://github.com/apache/ratis) with Maven 4:

```
mvn package -DskipTests
```

### Expected
Build succeeds (as it does with Maven 3). All properties resolve through the chain.

### Actual
```
[ERROR] 'distributionManagement.repository.[${distMgmtReleasesId}].id' contains an uninterpolated expression. @ line 37, column 7
[ERROR] 'distributionManagement.snapshotRepository.[${distMgmtSnapshotsId}].id' contains an uninterpolated expression. @ line 42, column 7
```

## Root Cause

Ratis defines a layer of indirection on top of the Apache parent POM's `distributionManagement` properties:

**Apache parent POM (`org.apache:apache:38`):**
```xml
<properties>
    <distMgmtReleasesId>apache.releases.https</distMgmtReleasesId>
</properties>
<distributionManagement>
    <repository>
        <id>${distMgmtReleasesId}</id>
    </repository>
</distributionManagement>
```

**Ratis POM overrides `distributionManagement` with staging properties:**
```xml
<properties>
    <distMgmtStagingId>${distMgmtReleasesId}</distMgmtStagingId>
</properties>
<distributionManagement>
    <repository>
        <id>${distMgmtStagingId}</id>
    </repository>
</distributionManagement>
```

The interpolation chain is: `${distMgmtStagingId}` → `${distMgmtReleasesId}` → `apache.releases.https`

Maven 4's validator runs after the first interpolation pass, sees `${distMgmtReleasesId}` (partially resolved), and rejects it. Maven 3 fully interpolates all nested properties before validation.

## Impact

Only affects projects that add an extra indirection layer over the Apache parent's `distributionManagement` properties. Currently only ratis is affected in our test suite of ~960 Apache repositories.

## Environment

- Maven: 4.0.0-SNAPSHOT (branch `maven-4.0.x-test-fixes`)
- Java: 17.0.18 (Eclipse Adoptium)

Test results: https://github.com/gnodet/maven4-testing/issues/9934

_Claude Code on behalf of Guillaume Nodet_

--- a/impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelValidator.java
+++ b/impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelValidator.java
@@ -696,14 +696,14 @@ public class DefaultModelValidator implements ModelValidator {
             DistributionManagement distMgmt = model.getDistributionManagement();
             if (distMgmt != null) {
                 validateRawRepository(
-                        problems, distMgmt.getRepository(), "distributionManagement.repository.", "", true, false);
+                        problems, distMgmt.getRepository(), "distributionManagement.repository.", "", true, true);
                 validateRawRepository(
                         problems,
                         distMgmt.getSnapshotRepository(),
                         "distributionManagement.snapshotRepository.",
                         "",
                         true,
-                        false);
+                        true);
             }
         }
     }

--- a/impl/maven-impl/src/test/java/org/apache/maven/impl/model/DefaultModelValidatorTest.java
+++ b/impl/maven-impl/src/test/java/org/apache/maven/impl/model/DefaultModelValidatorTest.java
@@ -929,18 +929,23 @@ class DefaultModelValidatorTest {
     void repositoryWithUninterpolatedId() throws Exception {
         SimpleProblemCollector result = validateRaw("raw-model/repository-with-uninterpolated-id.xml");
         // Uninterpolated expressions in repository IDs should cause validation errors
-        assertViolations(result, 0, 3, 0);
+        // distributionManagement repositories skip expression check since parent properties
+        // may not be available at file model validation stage
+        assertViolations(result, 0, 2, 0);
 
-        // Check that all three repository ID validation errors are present
+        // Check that repository ID validation errors are present for repositories and pluginRepositories
         assertTrue(result.getErrors().stream()
                 .anyMatch(error -> error.contains("repositories.repository.[${repository.id}].id")
                         && error.contains("contains an uninterpolated expression")));
         assertTrue(result.getErrors().stream()
                 .anyMatch(error -> error.contains("pluginRepositories.pluginRepository.[${plugin.repository.id}].id")
                         && error.contains("contains an uninterpolated expression")));
-        assertTrue(result.getErrors().stream()
-                .anyMatch(error -> error.contains("distributionManagement.repository.[${staging.repository.id}].id")
-                        && error.contains("contains an uninterpolated expression")));
+    }
+
+    @Test
+    void distributionManagementWithChainedPropertyInId() throws Exception {
+        SimpleProblemCollector result = validateRaw("raw-model/dm-with-chained-property-in-id.xml");
+        assertViolations(result, 0, 0, 0);
     }
 
     @Test

--- a/impl/maven-impl/src/test/resources/poms/validation/raw-model/dm-with-chained-property-in-id.xml
+++ b/impl/maven-impl/src/test/resources/poms/validation/raw-model/dm-with-chained-property-in-id.xml
@@ -1,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.apache.maven.validation</groupId>
+  <artifactId>project</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+
+  <properties>
+    <distMgmtStagingId>${distMgmtReleasesId}</distMgmtStagingId>
+    <distMgmtSnapshotsId>${distMgmtReleasesId}</distMgmtSnapshotsId>
+  </properties>
+
+  <distributionManagement>
+    <repository>
+      <id>${distMgmtStagingId}</id>
+      <url>https://repository.apache.org/service/local/staging/deploy/maven2</url>
+    </repository>
+    <snapshotRepository>
+      <id>${distMgmtSnapshotsId}</id>
+      <url>https://repository.apache.org/content/repositories/snapshots</url>
+    </snapshotRepository>
+  </distributionManagement>
+
+</project>

